numpy/core/src/umath/matmul.c.src

```diff
@@ -281,44 +281,45 @@ OBJECT_matmul_inner_noblas(void *_ip1, npy_intp is1_m, npy_intp is1_n,
     npy_intp ib2_p = is2_p * dp;
     npy_intp ob_p  = os_p * dp;
 
-    PyObject *tmp1, *tmp2, *tmp = NULL;
-    PyObject **tmp3;
+    PyObject *product, *sum_of_products = NULL;
 
     for (npy_intp m = 0; m < dm; m++) {
         for (npy_intp p = 0; p < dp; p++) {
-            if ( 0 == dn ) return;
+            if ( 0 == dn ) {
+                sum_of_products = PyLong_FromLong(0);
```

Comment:
I know it "should never happen", but you ought to check that this is not `NULL` too.